dsn.go

```diff
@@ -196,7 +196,8 @@ func (cfg *Config) FormatDSN() string {
 
 	// /dbname
 	buf.WriteByte('/')
-	buf.WriteString(cfg.DBName)
+	dbNameEncoded := url.QueryEscape(cfg.DBName)
```

Comment:
This should be PathEscape rather than QueryEscape.
QueryEscape will escape many characters that we don't want to escape.